chain/client/src/client.rs

```diff
@@ -197,17 +211,33 @@ impl Client {
             &self.runtime_adapter.get_epoch_id_from_prev_block(&head.last_block_hash).unwrap(),
             next_height,
         )?;
-        if !self.adv_produce_blocks {
+
+        #[allow(unused_assignments)]
+        #[allow(unused_mut)]
+        let mut adv_produce_blocks = false;
+
+        #[allow(unused_assignments)]
+        #[allow(unused_mut)]
+        let mut adv_produce_blocks_only_valid = false;
+
+        #[cfg(feature = "adversarial")]
+        {
+            adv_produce_blocks = self.adv_produce_blocks;
+            adv_produce_blocks_only_valid = self.adv_produce_blocks_only_valid;
+        }
+
+        if !adv_produce_blocks {
```

Comment:
This is too verbose. Why don't we just make the whole block feature-gated?

Reply:
The problem is that the code inside the local variable should be available in the non-adversarial environment. Adversaries will choose to run it, or not to run it, hence the whole dancing.

```rust
if !adv_produce_blocks { // adversaries choose if they want to execute the code inside.
    // This code here is non-adversarial
}
```

When compiled without the `adversarial` feature it is just the same as removing the condition and the compiler should remove it.

Reply:
As discussed offline added a `adversarial_variable` macro that wraps the action of creating the variable and its artifacts.